From 803c6568b529250fe674e4c5d7131e9033e3cf7f Mon Sep 17 00:00:00 2001
From: Jussi Kantell <jussi.kantell@intel.com>
Date: Mon, 29 Oct 2012 20:40:00 +0200
Subject: Dalvik: choose android os layer for x86-atom builds

BZ: 64734

This patch fixes the os layer selection in Dvm.mk which previously
used only dvmArch==arm logic to determine when the dalvik is build
for a device. Without this patch x86-atom devices are using a
stubbed out version of os layer which means in practice
that thread priority changes in java.lang.VMThread are not
doing anything.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I9a99884e4078e39c220e48685637574fda1786c8
Orig-MCG-Change-Id: I344d87f2874dcdc2bd74ce71afd76a35b8184db4
Signed-off-by: Jussi Kantell <jussi.kantell@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/Dvm.mk |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/vm/Dvm.mk b/vm/Dvm.mk
index 9644f47..cc36332 100644
--- a/vm/Dvm.mk
+++ b/vm/Dvm.mk
@@ -213,7 +213,11 @@ endif
 ifneq ($(filter arm mips,$(dvm_arch)),)
   LOCAL_SRC_FILES += os/android.cpp
 else
-  LOCAL_SRC_FILES += os/linux.cpp
+   ifeq ($(dvm_arch_variant),x86-atom)
+   LOCAL_SRC_FILES += os/android.cpp
+   else
+   LOCAL_SRC_FILES += os/linux.cpp
+   endif
 endif
 
 WITH_COPYING_GC := $(strip $(WITH_COPYING_GC))
-- 
1.7.4.1

